modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java

```diff
@@ -167,29 +173,29 @@ public void testRetrieveDefaultCorrelationLoggingConfigsTest() throws Exception
     dependsOnMethods = { "testRetrieveDefaultCorrelationLoggingConfigsTest" })
     public void testEnableAllCorrelationLoggingConfigsTest() throws Exception {
 
-        configureCorrelationLoggingComponent(new String[] {"http", "jdbc", "synapse", "ldap", "method-calls"}, true);
+        configureCorrelationLoggingComponent(new String[] {"http", "synapse", "ldap", "method-calls"}, true);
         log.info("***e UserMode:" + this.userMode);
-        InvokeTestAPI();
+
         // Validate Correlation Logs
-        // String logLine = bufferedReader.readLine();
-        // log.info("***e1: " + logLine);
-        // Thread.sleep(5000);
         resetAllLogs();
-        String logLine;
-        while ((logLine = bufferedReader.readLine()) != null) {
-            log.info("***e1: " + logLine);
-            assertTrue(isSynapseLogLine(logLine) || isHTTPLogLine(logLine) ||
-                    isJDBCLogLine(logLine) || isMethodCallsLogLine(logLine) || logLine.contains("Started log handler"));
-            if (logLine.contains(CORRELATION_ID)) {
-                correlationIDLog = true;
+
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            InvokeTestAPI();
```

Comment:
InvokeTestAPI(); >> Do we need this call to be made continuously?
So can't we get this outside the await block?

Reply:
Moved it to outside https://github.com/wso2/product-apim/pull/13088/commits/2c7b40a5f8ca5cab63ad8b5cb0d8bc511237cc5a